network/test/meshengine_test.go

```diff
@@ -320,7 +344,10 @@ func (suite *MeshEngineTestSuite) conduitCloseScenario(send ConduitSendWrapperFu
 	}
 
 	// allow nodes to heartbeat and discover each other
-	time.Sleep(2 * time.Second)
+	// each node will register ~D protect messages, where D is the default out-degree
+	for i := 0; i < pubsub.GossipSubD*count; i++ {
+		<-suite.obs
```

Comment:
```suggestion
                unittest.RequireCloseBefore(t, suite.obs, 2*time.Second, "could not register protect message on time")
```
better not to allow blocking code on tests without timeout.